ui/app/pages/confirm-transaction/conf-tx.js

```diff
@@ -45,188 +40,220 @@ function mapStateToProps (state) {
   }
 }
 
-inherits(ConfirmTxScreen, Component)
-function ConfirmTxScreen () {
-  Component.call(this)
-}
-
-ConfirmTxScreen.prototype.getUnapprovedMessagesTotal = function () {
-  const {
-    unapprovedMsgCount = 0,
-    unapprovedPersonalMsgCount = 0,
-    unapprovedTypedMessagesCount = 0,
-  } = this.props
-
-  return unapprovedTypedMessagesCount + unapprovedMsgCount + unapprovedPersonalMsgCount
-}
-
-ConfirmTxScreen.prototype.componentDidMount = function () {
-  const {
-    unapprovedTxs = {},
-    network,
-    send,
-  } = this.props
-  const unconfTxList = txHelper(unapprovedTxs, {}, {}, {}, network)
-
-  if (unconfTxList.length === 0 && !send.to && this.getUnapprovedMessagesTotal() === 0) {
-    this.props.history.push(DEFAULT_ROUTE)
+class ConfirmTxScreen extends Component {
+  static propTypes = {
+    unapprovedMsgCount: PropTypes.number,
+    unapprovedPersonalMsgCount: PropTypes.number,
+    unapprovedTypedMessagesCount: PropTypes.number,
+    network: PropTypes.number,
+    index: PropTypes.number,
+    unapprovedTxs: PropTypes.array,
+    unapprovedMsgs: PropTypes.array,
+    unapprovedPersonalMsgs: PropTypes.array,
+    unapprovedTypedMessages: PropTypes.array,
+    match: PropTypes.shape({
+      params: PropTypes.shape({
+        id: PropTypes.number,
+      }),
+    }),
+
+    selectedAddressTxList: PropTypes.array,
+    currentCurrency: PropTypes.string,
+    blockGasLimit: PropTypes.string,
+    conversionRate: PropTypes.number,
```

Comment:
Hmm, this prop is never passed in :thinking: 
I guess it's correct to add it here, since it is used, but maybe we should delete it instead.

Reply:
After further investigation, this and probably other props passed into `SignatureRequest` should be omitted, because they're retrieved separate in the SignatureRequest container component.
This can wait for a separate PR I suppose

Reply:
I see that the same applies for `accounts`, and `selectedAddress`

Reply:
`accounts` is only pulled from state for one of the two components returned from `signatureSelect`. I'll remove `selectedAddress` and `conversionRate` though, those are both clearly redundant (and not actually passed in, as you noticed 🤦‍♂).

Reply:
`accounts` isn't passed in either though, so if it is used in one of the `signatureSelect` components, it must not be very important, because it ain't there.

Reply:
Line 236 rn is `accounts={this.props.accounts}`, which passes it in?

Reply:
I mean it's not passed into _this_ component; it's not retrieved in `mapStateToProps`, nor is it passed in from above.

Reply:
Oh, heck. Good point.